service-registry/src/main/java/io/servicecomb/serviceregistry/client/http/RestUtils.java

```diff
@@ -126,16 +135,43 @@ public static RequestContext createRequestContext(HttpMethod method, IpPort ipPo
     return requestContext;
   }
 
+  public static SignRequest createSignRequest(RequestContext requestContext, String url, Map<String, String> headers) {
+    SignRequest signReq = new SignRequest();
+    StringBuilder endpoint = new StringBuilder("https://" + requestContext.getIpPort().getHostOrIp());
+    endpoint.append(":" + requestContext.getIpPort().getPort());
+    endpoint.append(url);
+    try {
+      signReq.setEndpoint(new URI(endpoint.toString()));
+    } catch (URISyntaxException e) {
+      LOGGER.error("set uri failed, uri is {}, message: {}", endpoint.toString(), e.getMessage());
+      return null;
```

Comment:
It's better not to return null, but set a null value to Endpoint. see examples null check problem